packages/compass-import-export/src/import/import-csv.spec.ts

```diff
@@ -780,6 +781,101 @@ describe('importCSV', function () {
       writeErrors: [],
     });
   });
+
+  it('does not mind windows style line breaks', async function () {
+    const text = await fs.promises.readFile(fixtures.csv.good_commas, 'utf8');
+    const replaced = text.replace(/\n/g, '\r\n');
+    const input = Readable.from(replaced);
+
+    const output = temp.createWriteStream();
+
+    const ns = 'db.col';
+    const fields = {
+      _id: 'string',
+      value: 'mixed',
+    } as const;
+
+    await importCSV({
+      dataService,
+      ns,
+      fields,
+      input,
+      output,
+      delimiter: ',',
+    });
+
+    const docs = await dataService.find(ns, {}, { promoteValues: false });
+
+    expect(docs).to.have.length(3);
+
+    for (const doc of docs) {
+      expect(Object.keys(doc)).to.deep.equal(['_id', 'value']);
+    }
+  });
+
+  it('does not mind if a file is not valid utf8', async function () {
+    const latin1Buffer = buffer.transcode(
```

Comment:
A bit tricky and I don't know if this is the best way to get a file that's in a different encoding, but it seems that import at least doesn't totally fall over in that case, so probably good enough?

Reply:
> but it seems that import at least doesn't totally fall over in that case, so probably good enough?

It will break if the file is in ASCII-incompatible encodings like UTF-16.

I think ideally we’d _want_ to break on invalid UTF-8 as input. I can’t say for sure but I’d expect that 99 % of the time users would rather have no data or only partial data imported rather than broken data.

Reply:
Can you think of a test for that and code to check for it?

Reply:
In terms of a test, the one here should work just fine. For checking the encoding, this should work:

```js
class Utf8Validator extends Transform {
  decoder = new TextDecoder('utf8', { fatal: true, ignoreBOM: true })

  _transform(chunk: Buffer, enc: unknown, cb: (err: null | Error, chunk?: Buffer) => void) {
    try {
      this.decoder.decode(chunk, { stream: true });
    } catch (err: any) {
      cb(err);
      return;
    }
    cb(null, chunk);
  }
}

Readable.from(Buffer.from('héllo', 'utf8')).pipe(new Utf8Validator()).pipe(process.stdout);
// Readable.from(Buffer.from('héllo', 'latin1')).pipe(new Utf8Validator()).pipe(process.stdout);
```

Reply:
Thanks! Updated now.